modules/database/src/ioc/db/dbChannel.c

```diff
@@ -450,26 +450,53 @@ dbChannel * dbChannelCreate(const char *name)
     const char *pname = name;
     DBENTRY dbEntry;
     dbChannel *chan = NULL;
-    char *cname;
+    char *cname = NULL;
     dbAddr *paddr;
     long status;
 
     if (!name || !*name || !pdbbase)
         return NULL;
 
+    /* If the timestamp filter is available, intercept requests for the TIME
+       field and replace the field with a channel filter string. */
+    {
+        static char const tsfilter[] = ".{\"ts\":{\"num\":\"dbl\"}}";
+        static int exists_ts = -1;
+
+        size_t namelen = strlen(name);
+        size_t reclen = namelen - 5; /* subtract strlen(".TIME"), MSVC won't optimize */
+        int has_time = reclen > 0 && !strcmp(name + reclen, ".TIME");
+
+        if (exists_ts == -1) {
+            /* Only look for the filter on first use. */
+            exists_ts = dbFindFilter("ts", 2) != NULL;
```

Comment:
This lazy initialization should be done in `dbChannelInit()`, which runs as part of `iocInit()`.  Well after filters are registered.